src/SFA.DAS.ApprenticeCommitments.UnitTests/Application/Services/TrainingProviderServiceTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Threading.Tasks;
+using AutoFixture.NUnit3;
+using FluentAssertions;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.ApprenticeCommitments.Apis.TrainingProviderApi;
+using SFA.DAS.ApprenticeCommitments.Application.Services;
+using SFA.DAS.ApprenticeCommitments.Configuration;
+using SFA.DAS.SharedOuterApi.Infrastructure;
+using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.ApprenticeCommitments.UnitTests.Application.Services
+{
+
+    [TestFixture]
+    public class TrainingProviderServiceTests
+    {
+        [Test, AutoData]
+        public async Task WhenNoTrainingProviderIsFound(
+            long trainingProviderId,
+            [Frozen] Mock<IInternalApiClient<TrainingProviderConfiguration>> client)
+        {
+            var sut = new TrainingProviderService(client.Object);
+            Func<Task> func = async () => await sut.GetTrainingProviderDetails(trainingProviderId);
+
+            func.Should().Throw<HttpRequestContentException>();
+        }
+
+        [Test, AutoData]
+        public async Task WhenMultipleTrainingProvidersAreFound(
+            long trainingProviderId,
+            TrainingProviderResponse[] results,
+            [Frozen] Mock<IInternalApiClient<TrainingProviderConfiguration>> client)
+        {
+            ClientReturnsSearchWith(client, results);
+            
+            var sut = new TrainingProviderService(client.Object);
+            
+            Func<Task> func = async () => await sut.GetTrainingProviderDetails(trainingProviderId);
+
+            func.Should().Throw<HttpRequestContentException>();
+        }
+
+        [Test, AutoData]
+        public async Task WhenASingleTrainingProvidersIsFound(
+            long trainingProviderId,
+            TrainingProviderResponse result,
```

Comment:
I like these tests, very clear what they are doing 💯

Reply:
I hate the syntax of teh Setup. Prefer to use write a Fixture, but these aren't liked. The only compromise I find it to put the setup code in private functions